pkg/operator/ceph/object/objectstore.go

```diff
@@ -775,8 +805,15 @@ func enableRGWDashboard(context *Context) error {
 		if err != nil {
 			logger.Errorf("failed to set user %q secretkey. %v", DashboardUser, err)
 		}
+		if mgr.FileBasedPasswordSupported(context.clusterInfo) {
+			if err := os.Remove(secretFile.Name()); err != nil {
```

Comment:
Why isn't this part in a `defer`?

Reply:
@leseb [comment](https://github.com/rook/rook/pull/7552#discussion_r613384963)